Commit a16027c2dbeb3aec8fe19707bce7f3c25b04cf3a breaks FindUdfread.cmake

<!--- Please fill out this template to the best of your ability. You can always edit this issue once you have created it. -->
<!--- Read the following link before you create a new problem report: https://kodi.wiki/view/HOW-TO:Submit_a_bug_report  -->
## Bug report
### Describe the bug
Here is a clear and concise description of what the problem is:
<!--- Provide a more detailed introduction to the issue itself, and why you consider it to be a bug -->
<!--- A bug report that is not clear will be closed -->
<!--- Put your text below this line -->

Commit a16027c2dbeb3aec8fe19707bce7f3c25b04cf3a blows away the buildudfread macro, which is still referenced in the current master version of the File.


## Expected Behavior
Here is a clear and concise description of what was expected to happen:
<!--- Tell us what should happen -->
<!--- Put your text below this line -->

Build should progress and function instead of crashing and burning at the call to buildudfread.

## Actual Behavior
<!--- Tell us what happens instead -->
<!--- Put your text below this line -->

Crash and burn:

```
CMake Error at cmake/modules/FindUdfread.cmake:37 (buildudfread):
  Unknown CMake command "buildudfread".
Call Stack (most recent call first):
  cmake/scripts/common/Macros.cmake:378 (find_package)
  cmake/scripts/common/Macros.cmake:426 (find_package_with_ver)
  CMakeLists.txt:285 (core_optional_dep)
```

## Possible Fix
<!--- Not obligatory, but suggest a fix or reason for the bug -->
<!--- Put your text below this line -->

Either readd the macro definition or remove it's use.

### To Reproduce
Steps to reproduce the behavior:
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->
<!--- Put your text below this line -->
1. 
2.
3.


### Debuglog
<!--- Put your text below this line -->
<!--- A debuglog is always mandatory when creating an issue. Provide one! -->
The debuglog can be found here:



### Screenshots 
Here are some links or screenshots to help explain the problem:
<!--- Put your text below this line -->



## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:
<!--- How has this bug affected you? What were you trying to accomplish? -->
<!--- Put your text below this line -->



### Your Environment
Used Operating system:
<!--- Include as many relevant details about the environment you experienced the bug in -->
<!--- Put your text below this line. Checkboxes can easily be ticked once issue is created -->
 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [ ] Linux
 - [ ] macOS
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name:
 - Kodi version:



<!--- End of this issue -->
*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -138,7 +138,6 @@ if(UNIX)
   option(ENABLE_INTERNAL_FSTRCMP "Enable internal fstrcmp?" OFF)
   option(ENABLE_INTERNAL_DAV1D "Enable internal dav1d?" OFF)
   option(ENABLE_INTERNAL_GTEST "Enable internal gtest?" OFF)
-  option(ENABLE_INTERNAL_UDFREAD "Enable internal udfread?" OFF)
 endif()
 
 # System options

--- a/cmake/modules/FindUdfread.cmake
+++ b/cmake/modules/FindUdfread.cmake
@@ -30,28 +30,17 @@ if(NOT TARGET ${APP_NAME_LC}::${CMAKE_FIND_PACKAGE_NAME})
     endif()
   endif()
 
-  # Check for existing UDFREAD. If version >= UDFREAD-VERSION file version, dont build
-  # A corner case, but if a linux/freebsd user WANTS to build internal udfread, build anyway
-  if((libudfread_VERSION VERSION_LESS ${${${CMAKE_FIND_PACKAGE_NAME}_MODULE}_VER} AND ENABLE_INTERNAL_UDFREAD) OR
-     ((CORE_SYSTEM_NAME STREQUAL linux OR CORE_SYSTEM_NAME STREQUAL freebsd) AND ENABLE_INTERNAL_UDFREAD))
-    buildudfread()
-  else()
-    if(TARGET libudfread::libudfread)
-      get_target_property(UDFREAD_LIBRARY_RELEASE libudfread::libudfread IMPORTED_LOCATION_RELWITHDEBINFO)
-      get_target_property(UDFREAD_INCLUDE_DIR libudfread::libudfread INTERFACE_INCLUDE_DIRECTORIES)
-    elseif(TARGET PkgConfig::libudfread)
-      # First item is the full path of the library file found
-      # pkg_check_modules does not populate a variable of the found library explicitly
-      list(GET libudfread_LINK_LIBRARIES 0 UDFREAD_LIBRARY_RELEASE)
+  if(TARGET libudfread::libudfread)
+    get_target_property(UDFREAD_LIBRARY libudfread::libudfread IMPORTED_LOCATION_RELWITHDEBINFO)
+    get_target_property(UDFREAD_INCLUDE_DIR libudfread::libudfread INTERFACE_INCLUDE_DIRECTORIES)
+  elseif(TARGET PkgConfig::libudfread)
+    # First item is the full path of the library file found
+    # pkg_check_modules does not populate a variable of the found library explicitly
+    list(GET libudfread_LINK_LIBRARIES 0 UDFREAD_LIBRARY)
 
-      get_target_property(UDFREAD_INCLUDE_DIR PkgConfig::libudfread INTERFACE_INCLUDE_DIRECTORIES)
-      set(UDFREAD_VERSION ${libudfread_VERSION})
-    endif()
+    get_target_property(UDFREAD_INCLUDE_DIR PkgConfig::libudfread INTERFACE_INCLUDE_DIRECTORIES)
+    set(UDFREAD_VERSION ${libudfread_VERSION})
   endif()
-
-  include(SelectLibraryConfigurations)
-  select_library_configurations(UDFREAD)
-  unset(UDFREAD_LIBRARIES)
 
   find_package_handle_standard_args(Udfread
                                     REQUIRED_VARS UDFREAD_LIBRARY UDFREAD_INCLUDE_DIR
@@ -59,13 +48,13 @@ if(NOT TARGET ${APP_NAME_LC}::${CMAKE_FIND_PACKAGE_NAME})
 
   if(UDFREAD_FOUND)
     # windows cmake config populated target
-    if(TARGET libudfread::libudfread AND NOT TARGET udfread_build)
+    if(TARGET libudfread::libudfread)
       add_library(${APP_NAME_LC}::${CMAKE_FIND_PACKAGE_NAME} ALIAS libudfread::libudfread)
       add_library(LIBRARY::${CMAKE_FIND_PACKAGE_NAME} ALIAS libudfread::libudfread)
       set_target_properties(libudfread::libudfread PROPERTIES
                                                    INTERFACE_COMPILE_DEFINITIONS HAS_UDFREAD)
     # pkgconfig populated target that is sufficient version
-    elseif(TARGET PkgConfig::libudfread AND NOT TARGET udfread_build)
+    elseif(TARGET PkgConfig::libudfread)
       add_library(${APP_NAME_LC}::${CMAKE_FIND_PACKAGE_NAME} ALIAS PkgConfig::libudfread)
       add_library(LIBRARY::${CMAKE_FIND_PACKAGE_NAME} ALIAS PkgConfig::libudfread)
       set_target_properties(PkgConfig::libudfread PROPERTIES

